presto-orc/src/test/java/io/prestosql/orc/TestReadBloomFilter.java

```diff
@@ -0,0 +1,81 @@
+ * limitations under the License.
+ */
+package io.prestosql.orc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+import io.prestosql.orc.TupleDomainOrcPredicate.ColumnReference;
+import io.prestosql.spi.predicate.NullableValue;
+import io.prestosql.spi.type.Type;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Iterables.cycle;
+import static com.google.common.collect.Iterables.limit;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.prestosql.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static io.prestosql.orc.OrcReader.MAX_BATCH_SIZE;
+import static io.prestosql.orc.OrcTester.Format.ORC_12;
+import static io.prestosql.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static io.prestosql.orc.OrcTester.MAX_BLOCK_SIZE;
+import static io.prestosql.orc.OrcTester.writeOrcColumnHive;
+import static io.prestosql.orc.metadata.CompressionKind.LZ4;
+import static io.prestosql.spi.predicate.TupleDomain.fromFixedValues;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestReadBloomFilter
+{
+    @Test
+    public void test()
+            throws Exception
+    {
+        Stream<Integer> writeValues = newArrayList(limit(cycle(ImmutableList.of(1, 1_000_000)), 30_000)).stream();
+
+        try (TempFile tempFile = new TempFile()) {
+            writeOrcColumnHive(tempFile.getFile(), ORC_12, LZ4, BIGINT, writeValues.iterator());
+
+            // without predicate a normal block will be created
+            try (OrcRecordReader recordReader = createCustomOrcRecordReader(tempFile, OrcPredicate.TRUE, BIGINT, MAX_BATCH_SIZE)) {
+                assertEquals(recordReader.nextBatch(), 1024);
+            }
+
+            // predicate for specific value within the min/max range, but not in the bloom filter
+            TupleDomainOrcPredicate<String> predicate = new TupleDomainOrcPredicate<>(
+                    fromFixedValues(ImmutableMap.of("bigint_0", NullableValue.of(BIGINT, 500_000L))),
+                    ImmutableList.of(new ColumnReference<>("bigint_0", 0, BIGINT)),
+                    true);
+
+            try (OrcRecordReader recordReader = createCustomOrcRecordReader(tempFile, predicate, BIGINT, MAX_BATCH_SIZE)) {
+                assertEquals(recordReader.nextBatch(), -1);
+            }
+        }
+    }
+
+    private static OrcRecordReader createCustomOrcRecordReader(TempFile tempFile, OrcPredicate predicate, Type type, int initialBatchSize)
+            throws IOException
+    {
+        OrcDataSource orcDataSource = new FileOrcDataSource(tempFile.getFile(), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true);
```

Comment:
Add a version with the bloom filter flag set to false so we can show that rows were filtered *because* of the bloom filter.